snow/validators/manager.go

```diff
@@ -22,25 +23,83 @@ var (
 	fmt.Stringer
 
-	// Add a subnet's validator set to the manager.
-	//
-	// If the subnet had previously registered a validator set, false will be
-	// returned and the manager will not be modified.
-	Add(subnetID ids.ID, set Set) bool
+	// Add a new staker to the subnet.
+	// Returns an error if:
+	// - [weight] is 0
+	// - [nodeID] is already in the validator set
+	// If an error is returned, the set will be unmodified.
+	AddStaker(subnetID ids.ID, nodeID ids.NodeID, pk *bls.PublicKey, txID ids.ID, weight uint64) error
+
+	// AddWeight to an existing staker to the subnet.
+	// Returns an error if:
+	// - [weight] is 0
+	// - [nodeID] is not already in the validator set
+	// If an error is returned, the set will be unmodified.
+	// AddWeight can result in a total weight that overflows uint64.
+	// In this case no error will be returned for this call.
+	// However, the next TotalWeight call will return an error.
+	AddWeight(subnetID ids.ID, nodeID ids.NodeID, weight uint64) error
+
+	// GetWeight retrieves the validator weight from the subnet.
+	GetWeight(subnetID ids.ID, validatorID ids.NodeID) uint64
+
+	// GetValidator returns the validator tied to the specified ID in subnet.
+	// If the validator doesn't exist, returns false.
+	GetValidator(subnetID ids.ID, validatorID ids.NodeID) (*Validator, bool)
+
+	// GetValidatoIDs returns the validator IDs in the subnet.
+	GetValidatorIDs(subnetID ids.ID) []ids.NodeID
+
+	// SubsetWeight returns the sum of the weights of the validators in the subnet.
+	// Returns err if subset weight overflows uint64.
+	SubsetWeight(subnetID ids.ID, validatorIDs set.Set[ids.NodeID]) (uint64, error)
 
-	// Get returns the validator set for the given subnet
-	// Returns false if the subnet doesn't exist
-	Get(ids.ID) (Set, bool)
+	// RemoveWeight from a staker in the subnet. If the staker's weight becomes 0, the staker
+	// will be removed from the subnet set.
+	// Returns an error if:
+	// - [weight] is 0
+	// - [nodeID] is not already in the subnet set
+	// - the weight of the validator would become negative
+	// If an error is returned, the set will be unmodified.
+	RemoveWeight(subnetID ids.ID, nodeID ids.NodeID, weight uint64) error
+
+	// Contains returns true if there is a validator with the specified ID
+	// currently in the subnet.
+	Contains(subnetID ids.ID, validatorID ids.NodeID) bool
+
+	// Len returns the number of validators currently in the subnet.
+	Len(subnetID ids.ID) int
+
+	// TotalWeight returns the cumulative weight of all validators in the subnet.
+	// Returns err if total weight overflows uint64.
+	TotalWeight(subnetID ids.ID) (uint64, error)
+
+	// Sample returns a collection of validatorIDs in the subnet, potentially with duplicates.
+	// If sampling the requested size isn't possible, an error will be returned.
+	Sample(subnetID ids.ID, size int) ([]ids.NodeID, error)
+
+	// Map of the validators in this subnet
+	GetMap(subnetID ids.ID) map[ids.NodeID]*GetValidatorOutput
```

Comment:
nit: I think we don't need to expose this anymore? We mostly use GetValidatorIDs